src/main/webapp/feed.html

```diff
@@ -21,11 +21,48 @@ <h1 class="title">🌏project title</h1>
         <div id="events-feed"></div>
         <div class="feed-right-side">
             <div class="feed-box create-event-box"><div class="feed-box-content create-event-box-content"><p class="icon">📅</p><p class="create-event-text">create an event</p></div></div>
-            <div class="feed-box challenges-box">
+            <div class="feed-box challenges-box" onclick="openChallengesModal()">
                 <div class="feed-box-content challenges-box-content">
                     <p class="icon">🌏</p><p class="challenges-text">challenges</p>
                 </div>
-                <div class="challenges-badge" id="challenges-badge"><p class="icon badge-icon" id="challenges-badge-icon"></p><p class="challenge-steps" id="challenge-steps"></p></div>
+                <div class="challenges-badge" id="challenges-badge">
+                    <p class="icon badge-icon" id="challenges-badge-icon"></p>
+                    <p class="challenge-steps" id="challenge-steps"></p>
+                    <div class="badge-filling" id="feed-badge-filling"></div>
+                </div>
+            </div>
+        </div>
+    </div>
+    <div id="challenges-modal">
+        <div id="challenges-nav-bar" class="challenges-modal-content">
+            <p id="challenges-nav-bar-header">Challenges</p>
+        </div>
+        <div id="challenges-main-panel" class="challenges-modal-content">
+            <p id="close-modal-button" onclick="closeChallengesModal()">x</p>
+            <div id="challenges-main-panel-upper-content">
+                <div id="challenges-main-panel-text" class="challenges-main-panel-text">
```

Comment:
Because challenges-main-panel-test is a class you wouldn't need div id=. Just keep the class